src/main/java/org/elasticsearch/percolator/PercolateContext.java

```diff
@@ -720,4 +721,14 @@ public SearchContext useSlowScroll(boolean useSlowScroll) {
     public Counter timeEstimateCounter() {
         throw new UnsupportedOperationException();
     }
+
+    @Override
+    public boolean requireDocsCollectedInOrder() {
+        return executeDocsInOrder;
+    }
+
+    @Override
+    public void setRequireDocsCollectedInOrder(boolean docsInOrder) {
+        this.executeDocsInOrder = docsInOrder;
```

Comment:
Maybe this should just be:

```java
    @Override
    public void setRequireDocsCollectedInOrder() {
        this.executeDocsInOrder = true;
    }
```

This way we would be sure that collection happens in order if any of the features that are used require in-order scoring? (Otherwise, you could override a previously set value.)